From 04371af7b55428a3dd3ffc27a337abbe78b3f234 Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Sat, 3 Sep 2011 17:03:59 +0200
Subject: [PATCH 6/6] Fix slash replacement for windows-style paths

---
 giscanner/utils.py |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/giscanner/utils.py b/giscanner/utils.py
index 2263b8c..91857e1 100644
--- a/giscanner/utils.py
+++ b/giscanner/utils.py
@@ -99,6 +99,7 @@ def extract_libtool(la_file):
     #        in a way which is compatible with both libtool 2.2
     #        and pre-2.2. Johan 2008-10-21
     libname = libname.replace('.libs/.libs', '.libs')
+    libname = libname.replace('.libs\\.libs', '.libs')
     return libname
 
 # Returns arguments for invoking libtool, if applicable, otherwise None
-- 
1.7.4.1

