--- src/Makefile.am	2011-08-30 01:16:14.670683737 +0200
+++ src/Makefile.am	2011-08-30 01:16:40.655940935 +0200
@@ -21,7 +21,7 @@
 
 libdvdnav_la_LDFLAGS = \
 	-version-info $(DVDNAV_LT_CURRENT):$(DVDNAV_LT_REVISION):$(DVDNAV_LT_AGE) \
-	-export-symbols-regex "(^dvdnav.*|^nav.*|^ifo.*|^DVD.*)"
+	-export-symbols-regex "(^dvdnav.*|^nav.*|^ifo.*|^DVD.*)" -ldvdread
 #	-release $(DVDNAV_MAJOR).$(DVDNAV_MINOR).$(DVDNAV_SUB)
 
 libdvdnavmini_la_SOURCES = $(libdvdnav_la_SOURCES)
