diff --git a/configure.ac b/configure.ac
index 681746d..34b1c7f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -59,12 +59,20 @@ AC_CHECK_FUNCS([memset], , AC_ERROR([memset not found in libc]))
 AC_CHECK_FUNCS([pow], , AC_CHECK_LIB(m, pow, , AC_ERROR([cannot find pow])) )
 AC_CHECK_LIB(z, zlibVersion, , AC_ERROR([zlib not installed]))
 
+AC_CANONICAL_HOST
+
 case $host_os in
   aix*)
    LIBPNG_DEFINES="-DPNG_CONFIGURE_LIBPNG -D_ALL_SOURCE";;
   *)
    LIBPNG_DEFINES=-DPNG_CONFIGURE_LIBPNG;;
 esac
+
+case "$host" in
+  *-*-mingw*)
+    LIBPNG_DEFINES="-DPNG_BUILD_DLL $LIBPNG_DEFINES";;
+esac
+
 LIBPNG_DEFINES=$LIBPNG_DEFINES
 AC_SUBST(LIBPNG_DEFINES)
 
