diff -ur pygobject-2.26.0//gi/Makefile.in pygobject-2.26.0.new//gi/Makefile.in
--- pygobject-2.26.0//gi/Makefile.in	2010-09-27 18:37:43.000000000 +0200
+++ pygobject-2.26.0.new//gi/Makefile.in	2011-09-04 14:24:11.000000000 +0200
@@ -644,9 +646,9 @@
 	done || exit $$?; \
 	if test -n "$$dlist"; then \
 	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(pygidir)" $$dlist; \
+	    PYTHON= $(py_compile) --basedir "$(pygidir)" $$dlist; \
 	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(pygidir)" $$dlist; \
+	    PYTHON= $(py_compile) --destdir "$(DESTDIR)" --basedir "$(pygidir)" $$dlist; \
 	  fi; \
 	else :; fi
 
diff -ur pygobject-2.26.0//gi/overrides/Makefile.in pygobject-2.26.0.new//gi/overrides/Makefile.in
--- pygobject-2.26.0//gi/overrides/Makefile.in	2010-09-27 18:37:43.000000000 +0200
+++ pygobject-2.26.0.new//gi/overrides/Makefile.in	2011-09-04 14:24:11.000000000 +0200
@@ -303,9 +305,9 @@
 	done || exit $$?; \
 	if test -n "$$dlist"; then \
 	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(pygioverridesdir)" $$dlist; \
+	    PYTHON= $(py_compile) --basedir "$(pygioverridesdir)" $$dlist; \
 	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(pygioverridesdir)" $$dlist; \
+	    PYTHON= $(py_compile) --destdir "$(DESTDIR)" --basedir "$(pygioverridesdir)" $$dlist; \
 	  fi; \
 	else :; fi
 
diff -ur pygobject-2.26.0//gi/repository/Makefile.in pygobject-2.26.0.new//gi/repository/Makefile.in
--- pygobject-2.26.0//gi/repository/Makefile.in	2010-09-27 18:37:43.000000000 +0200
+++ pygobject-2.26.0.new//gi/repository/Makefile.in	2011-09-04 14:24:11.000000000 +0200
@@ -298,9 +300,9 @@
 	done || exit $$?; \
 	if test -n "$$dlist"; then \
 	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(pygirepositorydir)" $$dlist; \
+	    PYTHON= $(py_compile) --basedir "$(pygirepositorydir)" $$dlist; \
 	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(pygirepositorydir)" $$dlist; \
+	    PYTHON= $(py_compile) --destdir "$(DESTDIR)" --basedir "$(pygirepositorydir)" $$dlist; \
 	  fi; \
 	else :; fi
 
diff -ur pygobject-2.26.0//gio/Makefile.am pygobject-2.26.0.new//gio/Makefile.am
--- pygobject-2.26.0//gio/Makefile.am	2010-09-14 17:53:21.000000000 +0200
+++ pygobject-2.26.0.new//gio/Makefile.am	2011-09-04 14:13:10.000000000 +0200
@@ -27,7 +27,8 @@
 # linker flags
 common_ldflags = -module -avoid-version
 if PLATFORM_WIN32
-common_ldflags += -no-undefined
+PYVER= `echo $(PYTHON_VERSION) | sed 's/\.//g'`
+common_ldflags += -Wl,$(pyexecdir)/../../libs/libpython$(PYVER).a -no-undefined
 endif
 
 CLEANFILES =
diff -ur pygobject-2.26.0//gio/Makefile.in pygobject-2.26.0.new//gio/Makefile.in
--- pygobject-2.26.0//gio/Makefile.in	2010-09-27 18:37:43.000000000 +0200
+++ pygobject-2.26.0.new//gio/Makefile.in	2011-09-04 14:24:11.000000000 +0200
@@ -35,7 +35,7 @@
 POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
-@PLATFORM_WIN32_TRUE@am__append_1 = -no-undefined
+@PLATFORM_WIN32_TRUE@am__append_1 = -Wl,$(pyexecdir)/../../libs/libpython$(PYVER).a -no-undefined
 @BUILD_GIO_TRUE@am__append_2 = _gio.la
 @BUILD_GIO_TRUE@am__append_3 = $(GIO_DEFS)
 @BUILD_GIOUNIX_TRUE@am__append_4 = unix.la
@@ -315,6 +317,7 @@
 
 # linker flags
 common_ldflags = -module -avoid-version $(am__append_1)
+@PLATFORM_WIN32_TRUE@PYVER = `echo $(PYTHON_VERSION) | sed 's/\.//g'`
 CLEANFILES = gio.c unix.c
 EXTRA_DIST = $(GIO_DEFS) $(GIO_OVERRIDES) $(GIOUNIX_DEFS) \
 	unix.override
@@ -566,9 +569,9 @@
 	done || exit $$?; \
 	if test -n "$$dlist"; then \
 	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(pygiodir)" $$dlist; \
+	    PYTHON= $(py_compile) --basedir "$(pygiodir)" $$dlist; \
 	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(pygiodir)" $$dlist; \
+	    PYTHON= $(py_compile) --destdir "$(DESTDIR)" --basedir "$(pygiodir)" $$dlist; \
 	  fi; \
 	else :; fi
 
diff -ur pygobject-2.26.0//glib/Makefile.am pygobject-2.26.0.new//glib/Makefile.am
--- pygobject-2.26.0//glib/Makefile.am	2010-09-14 17:53:21.000000000 +0200
+++ pygobject-2.26.0.new//glib/Makefile.am	2011-09-04 14:13:10.000000000 +0200
@@ -16,10 +16,12 @@
 
 common_ldflags = -module -avoid-version
 if PLATFORM_WIN32
-common_ldflags += -no-undefined
+PYVER= `echo $(PYTHON_VERSION) | sed 's/\.//g'`
+common_ldflags += -Wl,$(pyexecdir)/../../libs/libpython$(PYVER).a -no-undefined
 endif
 
 libpyglib_2_0_@PYTHON_BASENAME@_la_CFLAGS = $(GLIB_CFLAGS)
+libpyglib_2_0_@PYTHON_BASENAME@_la_LDFLAGS = $(common_ldflags)
 libpyglib_2_0_@PYTHON_BASENAME@_la_LIBADD = $(GLIB_LIBS) $(FFI_LIBS)
 libpyglib_2_0_@PYTHON_BASENAME@_la_SOURCES = 	\
 	pyglib.c		\
diff -ur pygobject-2.26.0//glib/Makefile.in pygobject-2.26.0.new//glib/Makefile.in
--- pygobject-2.26.0//glib/Makefile.in	2010-09-27 18:37:43.000000000 +0200
+++ pygobject-2.26.0.new//glib/Makefile.in	2011-09-04 14:24:11.000000000 +0200
@@ -34,7 +34,7 @@
 POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
-@PLATFORM_WIN32_TRUE@am__append_1 = -no-undefined
+@PLATFORM_WIN32_TRUE@am__append_1 = -Wl,$(pyexecdir)/../../libs/libpython$(PYVER).a -no-undefined
 @PLATFORM_WIN32_TRUE@am__append_2 = -DPLATFORM_WIN32
 subdir = glib
 DIST_COMMON = $(pkginclude_HEADERS) $(pyglib_PYTHON) \
@@ -98,7 +98,7 @@
 libpyglib_2_0_@PYTHON_BASENAME@_la_LINK = $(LIBTOOL) $(AM_V_lt) \
 	--tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=link \
 	$(CCLD) $(libpyglib_2_0_@PYTHON_BASENAME@_la_CFLAGS) $(CFLAGS) \
-	$(AM_LDFLAGS) $(LDFLAGS) -o $@
+	$(libpyglib_2_0_@PYTHON_BASENAME@_la_LDFLAGS) $(LDFLAGS) -o $@
 DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
@@ -301,7 +303,9 @@
 
 pyglib_LTLIBRARIES = _glib.la
 common_ldflags = -module -avoid-version $(am__append_1)
+@PLATFORM_WIN32_TRUE@PYVER = `echo $(PYTHON_VERSION) | sed 's/\.//g'`
 libpyglib_2_0_@PYTHON_BASENAME@_la_CFLAGS = $(GLIB_CFLAGS)
+libpyglib_2_0_@PYTHON_BASENAME@_la_LDFLAGS = $(common_ldflags)
 libpyglib_2_0_@PYTHON_BASENAME@_la_LIBADD = $(GLIB_LIBS) $(FFI_LIBS)
 libpyglib_2_0_@PYTHON_BASENAME@_la_SOURCES = \
 	pyglib.c		\
@@ -566,9 +570,9 @@
 	done || exit $$?; \
 	if test -n "$$dlist"; then \
 	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(pyglibdir)" $$dlist; \
+	    PYTHON= $(py_compile) --basedir "$(pyglibdir)" $$dlist; \
 	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(pyglibdir)" $$dlist; \
+	    PYTHON= $(py_compile) --destdir "$(DESTDIR)" --basedir "$(pyglibdir)" $$dlist; \
 	  fi; \
 	else :; fi
 
diff -ur pygobject-2.26.0//gobject/Makefile.am pygobject-2.26.0.new//gobject/Makefile.am
--- pygobject-2.26.0//gobject/Makefile.am	2010-09-14 17:53:21.000000000 +0200
+++ pygobject-2.26.0.new//gobject/Makefile.am	2011-09-04 14:13:10.000000000 +0200
@@ -16,7 +16,8 @@
 
 common_ldflags = -module -avoid-version
 if PLATFORM_WIN32
-common_ldflags += -no-undefined
+PYVER= `echo $(PYTHON_VERSION) | sed 's/\.//g'`
+common_ldflags += -Wl,$(pyexecdir)/../../libs/libpython$(PYVER).a -no-undefined
 endif
 
 _gobject_la_CFLAGS = \
diff -ur pygobject-2.26.0//gobject/Makefile.in pygobject-2.26.0.new//gobject/Makefile.in
--- pygobject-2.26.0//gobject/Makefile.in	2010-09-27 18:37:43.000000000 +0200
+++ pygobject-2.26.0.new//gobject/Makefile.in	2011-09-04 14:24:11.000000000 +0200
@@ -34,7 +34,7 @@
 POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
-@PLATFORM_WIN32_TRUE@am__append_1 = -no-undefined
+@PLATFORM_WIN32_TRUE@am__append_1 = -Wl,$(pyexecdir)/../../libs/libpython$(PYVER).a -no-undefined
 @ENABLE_INTROSPECTION_TRUE@am__append_2 = $(GI_CFLAGS)
 @HAVE_LIBFFI_TRUE@am__append_3 = ffi-marshaller.c ffi-marshaller.h
 @PLATFORM_WIN32_TRUE@am__append_4 = -DPLATFORM_WIN32
@@ -301,6 +303,7 @@
 
 pygobject_LTLIBRARIES = _gobject.la 
 common_ldflags = -module -avoid-version $(am__append_1)
+@PLATFORM_WIN32_TRUE@PYVER = `echo $(PYTHON_VERSION) | sed 's/\.//g'`
 _gobject_la_CFLAGS = -I$(top_srcdir)/glib -I$(top_srcdir)/gi \
 	$(PYTHON_INCLUDES) $(FFI_CFLAGS) $(GLIB_CFLAGS) \
 	-DPY_SSIZE_T_CLEAN $(am__append_2) $(am__append_4)
@@ -528,9 +531,9 @@
 	done || exit $$?; \
 	if test -n "$$dlist"; then \
 	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(pygobjectdir)" $$dlist; \
+	    PYTHON= $(py_compile) --basedir "$(pygobjectdir)" $$dlist; \
 	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(pygobjectdir)" $$dlist; \
+	    PYTHON= $(py_compile) --destdir "$(DESTDIR)" --basedir "$(pygobjectdir)" $$dlist; \
 	  fi; \
 	else :; fi
 
diff -ur pygobject-2.26.0//Makefile.in pygobject-2.26.0.new//Makefile.in
--- pygobject-2.26.0//Makefile.in	2010-09-27 18:37:44.000000000 +0200
+++ pygobject-2.26.0.new//Makefile.in	2011-09-04 14:24:11.000000000 +0200
@@ -497,9 +499,9 @@
 	done || exit $$?; \
 	if test -n "$$dlist"; then \
 	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(pkgpyexecdir)" $$dlist; \
+	    PYTHON= $(py_compile) --basedir "$(pkgpyexecdir)" $$dlist; \
 	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(pkgpyexecdir)" $$dlist; \
+	    PYTHON= $(py_compile) --destdir "$(DESTDIR)" --basedir "$(pkgpyexecdir)" $$dlist; \
 	  fi; \
 	else :; fi
 
@@ -535,9 +537,9 @@
 	done || exit $$?; \
 	if test -n "$$dlist"; then \
 	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(pyexecdir)" $$dlist; \
+	    PYTHON= $(py_compile) --basedir "$(pyexecdir)" $$dlist; \
 	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(pyexecdir)" $$dlist; \
+	    PYTHON= $(py_compile) --destdir "$(DESTDIR)" --basedir "$(pyexecdir)" $$dlist; \
 	  fi; \
 	else :; fi
 
--- pygobject-2.26.0/codegen/Makefile.in	2011-09-04 14:48:00.000000000 +0200
+++ pygobject-2.26.0/codegen/Makefile.in	2011-09-04 14:49:07.000000000 +0200
@@ -394,9 +394,9 @@
 	done || exit $$?; \
 	if test -n "$$dlist"; then \
 	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(codegendir)" $$dlist; \
+	    PYTHON= $(py_compile) --basedir "$(codegendir)" $$dlist; \
 	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(codegendir)" $$dlist; \
+	    PYTHON= $(py_compile) --destdir "$(DESTDIR)" --basedir "$(codegendir)" $$dlist; \
 	  fi; \
 	else :; fi
 
