--- win32/Makefile.gcc	2010-04-19 03:35:55.000000000 +0200
+++ win32/Makefile.gcc	2011-08-03 11:23:50.000000000 +0200
@@ -25,18 +25,21 @@
 
 STATICLIB = libz.a
 SHAREDLIB = zlib1.dll
-IMPLIB    = libzdll.a
+IMPLIB    = libz.dll.a
+
+BINARY_PATH = /usr/bin
+LIBRARY_PATH = /usr/lib
+INCLUDE_PATH = /usr/include
 
 #
 # Set to 1 if shared object needs to be installed
 #
-SHARED_MODE=0
+SHARED_MODE=1
 
 #LOC = -DASMV
 #LOC = -DDEBUG -g
 
 PREFIX =
-CC = $(PREFIX)gcc
 CFLAGS = $(LOC) -O3 -Wall
 EXTRA_CFLAGS = -DNO_VIZ
 
@@ -49,7 +52,7 @@
 AR = $(PREFIX)ar
 ARFLAGS = rcs
 
-RC = $(PREFIX)windres
+RC = $(WINDRES)
 RCFLAGS = --define GCC_WINDRES
 
 STRIP = $(PREFIX)strip
@@ -84,6 +87,7 @@
 
 $(STATICLIB): $(OBJS) $(OBJA)
 	$(AR) $(ARFLAGS) $@ $(OBJS) $(OBJA)
+	$(RANLIB) $(STATICLIB)
 
 $(IMPLIB): $(SHAREDLIB)
 
